noggin/controller/user.py

```diff
@@ -52,26 +54,40 @@ def user(ipa, username):
     )
 
 
-def _user_mod(ipa, form, username, details, redirect_to):
+def _user_mod(ipa, form, user, details, redirect_to):
     with handle_form_errors(form):
         try:
-            ipa.user_mod(username, **details)
+            ipa.user_mod(user.username, **details)
         except python_freeipa.exceptions.BadRequest as e:
             if e.message == 'no modifications to be performed':
                 raise FormError("non_field_errors", e.message)
             else:
                 app.logger.error(
-                    f'An error happened while editing user {username}: {e.message}'
+                    f'An error happened while editing user {user.username}: {e.message}'
                 )
                 raise FormError("non_field_errors", e.message)
         flash(
             Markup(
-                f'Profile Updated: <a href=\"{url_for("user", username=username)}\">'
+                f'Profile Updated: <a href=\"{url_for("user", username=user.username)}\">'
                 'view your profile</a>'
             ),
             'success',
         )
-        return redirect(url_for(redirect_to, username=username))
+        updated_user = User(user_or_404(ipa, user.username))
```

Comment:
for some reason, the ipa.user_mod seems to return some different values -- always getting the last_password_change as a changed field here. that is why i resorted to another api call here.

Reply:
Huh, that's wierd, I'll investigate this.

Reply:
OK I found why. The `user_mod` call  was not returning all fields, so the `last_password`change` was showing up as changed because it was empty in the new object.
I've added the `all=True` parameter to the call so we get all fields, and it worked.